pkg/models/context.go

```diff
@@ -40,7 +40,10 @@ func (ctx *ReqContext) Handle(cfg *setting.Cfg, status int, title string, err er
 }
 
 func (ctx *ReqContext) IsApiRequest() bool {
-	return strings.HasPrefix(ctx.Req.URL.Path, "/api")
+	return strings.HasPrefix(ctx.Req.URL.Path, "/api") ||
+		strings.HasPrefix(ctx.Req.URL.Path, "/alertmanager/") ||
+		strings.HasPrefix(ctx.Req.URL.Path, "/ruler/") ||
+		strings.HasPrefix(ctx.Req.URL.Path, "/prometheus/")
```

Comment:
This is required for returning 401 instead of redirecting to login page:
https://github.com/grafana/grafana/blob/ec12b344df3680b398acde4c5c6556b0de73321d/pkg/middleware/auth.go#L32-L40

I don't really like it though, I would prefer if the alerting API routes would be prefixed with `/api` (that would require some additional handling for removing this prefix when forwarding to lotex backends)

Reply:
Good call - I'm fine with prefixing everything with `/api`.